lisp/ess-inf.el

```diff
@@ -1312,6 +1312,14 @@ STRING). In all other cases the behavior is as described in
       string
     (concat string "\n")))
 
+(defun ess--delete-trailing-operator-maybe (input)
+  (when ess-eval-without-trailing-operator
```

Comment:
This will fail if ess-eval-without-trailing-operator is nil.  Presend functions must return a string.